File: Project Academy/09. Room/java/app/src/main/java/com/dicoding/academies/data/source/local/entity/CourseWithModule.java
package com.dicoding.academies.data.source.local.entity;

import androidx.room.Embedded;
import androidx.room.Relation;

import java.util.List;

public class CourseWithModule {
    @Embedded
    public CourseEntity mCourse;

    @Relation(parentColumn = "courseId", entityColumn = "courseId")
    public List<ModuleEntity> mModules;
}

File: Project Academy/12. Pagination-test/kotlin/app/src/main/java/com/dicoding/academies/data/source/local/entity/CourseWithModule.kt
package com.dicoding.academies.data.source.local.entity

import androidx.room.Embedded
import androidx.room.Relation

data class CourseWithModule (
    @Embedded
    var mCourse: CourseEntity,

    @Relation(parentColumn = "courseId", entityColumn = "courseId")
    var mModules: List<ModuleEntity>
)